tests/components/mqtt/__init__.py

```diff
@@ -0,0 +1 @@
+"""Tests for MQTT component."""
```

Comment:
It is. PyTest got confused collecting tests because `test_init.py` exists for both MQTT and Recorder. By adding `__init__.py` files I could resolve this. However you can't add a file without adding documentation otherwise pep257 complains, hence the comment.